src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -1365,15 +1365,15 @@ internal IEnumerable<SecurityKey> GetContentEncryptionKeys(JsonWebToken jwtToken
                         jwtToken.TryGetHeaderValue(JwtHeaderParameterNames.Apu, out string apu);
                         jwtToken.TryGetHeaderValue(JwtHeaderParameterNames.Apv, out string apv);
                         SecurityKey kdf = ecdhKeyExchangeProvider.GenerateKdf(apu, apv);
-                        var kwp = key.CryptoProviderFactory.CreateKeyWrapProviderForUnwrap(kdf, ecdhKeyExchangeProvider.GetEncryptionAlgorithm());
+                        using var kwp = key.CryptoProviderFactory.CreateKeyWrapProviderForUnwrap(kdf, ecdhKeyExchangeProvider.GetEncryptionAlgorithm());
```

Comment:
I don't think this is the right change - we shouldn't dispose here.

`CryptoProviderFactory.CreateKeyWrapProviderForUnwrap` ends up calling [`CustomCryptoProvider.Create`](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/c75a8f293899ceb6d8b334623d9a30c340b58370/src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs#L244) if it was set. `CustomCryptoProvider` implements `ICryptoProvider` which has a `Create` and a [`Release`](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/c75a8f293899ceb6d8b334623d9a30c340b58370/src/Microsoft.IdentityModel.Tokens/ICryptoProvider.cs#L34). So the interface intends for the creator to release the provider.

The more appropriate change would probably be to call [`CryptoProviderFactory.ReleaseKeyWrapProvider`](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/c75a8f293899ceb6d8b334623d9a30c340b58370/src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs#L824) maybe in the `finally`.